python/ray/tune/trial.py

```diff
@@ -404,15 +404,20 @@ def uses_placement_groups(self):
         return bool(self.placement_group_factory)
 
     def reset(self):
+        trainable_cls = self.get_trainable_cls()
```

Comment:
Nit: I would move the reset logic closer, remove the negation, and add a comment to explain this interaction.

```python
# Unset resources and placement_group_factory if there is a default_resource_request.
# The new Trial will be configured using the default_resource_request.
trainable_cls = self.get_trainable_cls()
clear_resources = (trainable_cls and
                   trainable_cls.default_resource_request(self.config))
resources = None if clear_resources else self.resources 
placement_group_factory = None if clear_resources else self.placement_group_factory

...

resources=resources
placement_group_factory=placement_group_factory
```